instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/RequestMetricsFactory.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import io.opentelemetry.api.metrics.Meter;
+
+@FunctionalInterface
+public interface RequestMetricsFactory {
```

Comment:
Possibly this class could be called `RequestMetrics` and the other `RequestListener`

Reply:
Note: if we ever want to implement [this comment](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2596#discussion_r603924647): we could encapsulate all span operations in a `RequestListener`

Reply:
Yeah that seems like a nice idea